src/screens/PostScreen.tsx

```diff
@@ -1,24 +1,45 @@
-// import { useState } from 'react';
-import { Image, Text, View } from 'react-native';
+import { useEffect, useState } from 'react';
+import { Image, Pressable, Text, View } from 'react-native';
+import supabase from '@/supabase/client';
 import HeartIcon from '../../assets/heart-icon.svg';
 import ShareIcon from '../../assets/messenger-icon.svg';
 import ProfilePlaceholder from '../../assets/profile-placeholder-icon.svg';
-// import supabase from '../../supabase/client';
 import { styles } from './styles';
 
 export default function PostScreen() {
-  // const [postData, setPostData] = useState({
-  //   description: 'description',
-  //   username: 'username',
-  //   imageURL: 'imageURL',
-  //   likes: 0,
-  // });
+  const [postData, setPostData] = useState({
+    created_at: 'created_at',
+    description: 'description',
+    image_url: 'image_url',
+    like_count: 0,
+    username: 'username',
+    uuid: 'uuid',
```

Comment:
I'm assuming these are placeholders that appear while the supabase query loads? If so, we can instead create a `isLoading` or `postData == null` condition that only renders `postData` after the data is fetched. You can use an inline if condition (`<condition> ? <then-body> : <else-body>`) to render either the postData or a loading screen.

Reply:
Additionally, I'd consider creating a `Post` type so these fields and their corresponding types are explicitly declared.